ui/raidboss/emulator/data/PopupTextAnalysis.js

```diff
@@ -208,11 +212,15 @@ export default class PopupTextAnalysis extends StubbedPopupText {
   }
 
   _createTextFor(text, textType, lowerTextKey, duration) {
-    // No-op
+    // No-op for functionality, but store off this info for feedback
+    this.currentTriggerStatus.responseType = textType;
+    this.currentTriggerStatus.responseLabel = text;
   }
 
   _playAudioFile(url, volume) {
-    // No-op
+    // No-op for functionality, but store off this info for feedback
+    this.currentTriggerStatus.responseType = 'audiofile';
```

Comment:
From code inspection, I'm sort of surprised why this doesn't override the `_createTextFor` when there is the normal audio alert + text on screen.

Reply:
I'm not sure I understand this comment?

`_createTextFor` is overridden right above this to handle info/alert/alarm but checking for if there's an audio file queued to play requires access to the `triggerHelper` which isn't available in that function.

Reply:
Sorry, this was just a drive-by "trying to understand the code without doing much debugging".

I guess my confusion was that even for an `alertText`, a sound file gets played via `_playAudioFile`, and so what prevents `this.currentTriggerStatus.responseType = 'audiofile';` from clobbering the value set in `this.currentTriggerStatus.responseType = textType;` from `_createTextFor`, as `_playAudioFile` runs after `_createTextFor`.

Reply:
Currently, nothing. There doesn't seem to be a good way to display both audio and text info within the confines of the button lines. I guess the icon could be the audio file icon, the border color could go to blue to indicate, but it could keep the text displayed?

Functionally this would be just removing the `this.currentTriggerStatus.responseLabel = url;` assignment.

Reply:
Sorry, maybe I'm only creating more confusion about my confusion.  What I'm trying to ask is, why does an `infoText` end up with `responseType === 'info'` and not `responseType === 'audiofile'` since `infoText` also plays a sound.  Your code appears to do what it should (the former) but I don't see from reading this why it's not doing the latter.

Sorry for the weird question, I was hoping you could clean up whatever misconception I was having so I didn't have to go do a bunch of debugging to understand why things work.

Reply:
Looks like this was a side-effect of https://github.com/quisquous/cactbot/commit/e0c67d03e98361ebb75400a3c128b987bbf351ae, `audioAllowed` was changed to `AudioAllowed` but not updated in `raidemulator.js` to reflect that change.

The `Options` global in `raidemulator.js` should probably be updated to just import from `raidboss_options.js`.

That will result in the expected behavior here (e.g. everything plays an audio alert so it all gets overwritten by the audio alert display), see screenshot below.

I think that in this case it doesn't really make sense to show the audio alert for default audio alerts? Functionally, in `_playAudioFile` check that the url matches the global `Options` entries and if so then don't overwrite the text display?

In addition, it probably doesn't make sense to overwrite the text for audio alerts, maybe it should just overwrite icon/color but leave the text the same (if there's text) since the audio file played can be pulled from details if needed.

![image](https://user-images.githubusercontent.com/6119598/113490502-62e44900-9498-11eb-9e8d-4dc116ec4f81.png)

Reply:
Aha!  Thank you for figuring it out.  Definitely agreed that text and icon should not take precedence if it is the default.  I think the previous behavior was good.

I'm not sure what the best way to do this functionally is.  I think somebody could write a trigger that plays the alert sound manually for a trigger but nothing else, so I'm a little bit skeptical of url checking.

Maybe `triggerHelper` needs to change the field it sets from `soundUrl` to `implicitSoundUrl` (or default? or auto? or some word?) and then you can figure out at the time of playing a sound whether or not the sound url is explicit or implicit.  You could have some `_playImplicitAudioFile` that the emulator could override to nothing?

Reply:
Review the latest commit's logic in `PopupTextAnalysis.js`'s `_playAudioFile` function. Also swapped the `raidemulator.js` `Options` property to pull from `raidboss_options.js`.